contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,339 @@
+
+        emit ReservesThresholdUpdate(
+            oldReservesThreshold,
+            newReservesThreshold
+        );
+    }
+
+    /// @notice helper function to set the surplus target
+    function _setSurplusTarget(IPCVDeposit newSurplusTarget) internal {
+        require(
+            address(newSurplusTarget) != address(0),
+            "PegStabilityModule: Invalid new surplus target"
+        );
+        require(
+            newSurplusTarget.balanceReportedIn() == address(underlyingToken),
+            "PegStabilityModule: Underlying token mismatch"
+        );
+
+        IPCVDeposit oldTarget = surplusTarget;
+        surplusTarget = newSurplusTarget;
+
+        emit SurplusTargetUpdate(oldTarget, newSurplusTarget);
+    }
+
+    // ----------- Public State Changing API -----------
+
+    /// @notice send any surplus reserves to the PCV allocation
+    function allocateSurplus() external override {
+        int256 currentSurplus = reservesSurplus();
+        require(
+            currentSurplus > 0,
+            "PegStabilityModule: No surplus to allocate"
+        );
+
+        _allocate(currentSurplus.toUint256());
+    }
+
+    /// @notice function to receive ERC20 tokens from external contracts
+    function deposit() external override {
+        int256 currentSurplus = reservesSurplus();
+        if (currentSurplus > 0) {
+            _allocate(currentSurplus.toUint256());
+        }
+    }
+
+    /// @notice function to redeem VOLT for an underlying asset
+    function redeem(
+        address to,
+        uint256 amountVoltIn,
+        uint256 minAmountOut
+    ) external virtual override whenNotPaused returns (uint256 amountOut) {
+        amountOut = getRedeemAmountOut(amountVoltIn);
+        require(
+            amountOut >= minAmountOut,
+            "PegStabilityModule: Redeem not enough out"
+        );
+
+        _beforeVoltRedeem(to, amountVoltIn, minAmountOut);
+
+        IERC20(volt()).safeTransferFrom(
+            msg.sender,
+            address(this),
+            amountVoltIn
+        );
+
+        SafeERC20.safeTransfer(underlyingToken, to, amountOut);
+
+        emit Redeem(to, amountVoltIn, amountOut);
+
+        _afterVoltRedeem(to, amountVoltIn, minAmountOut);
+    }
+
+    /// @notice function to buy VOLT for an underlying asset
+    function mint(
+        address to,
+        uint256 amountIn,
+        uint256 minAmountVoltOut
+    ) external virtual override whenNotPaused returns (uint256 amountVoltOut) {
+        amountVoltOut = getMintAmountOut(amountIn);
+        require(
+            amountVoltOut >= minAmountVoltOut,
+            "PegStabilityModule: Mint not enough out"
+        );
+
+        _beforeVoltMint(to, amountIn, minAmountVoltOut);
+
+        SafeERC20.safeTransferFrom(
+            underlyingToken,
+            msg.sender,
+            address(this),
+            amountIn
+        );
+
+        uint256 amountVoltToTransfer = Math.min(
+            volt().balanceOf(address(this)),
```

Comment:
Interesting, the use of `Math.min()` here could effectively mean that the `require(amountVoltOut >=minAmountVoltOut)` check above is bypassed. 

If the VOLT balance of the PSM is less than the `amountVoltOut` requested, then the user will be short changed and get less VOLT than they should do. Why not just revert if there's not enough VOLT on the PSM?

Reply:
ahh yeah good catch, I think this is because this behaviour was lifted from when the PSM could still mint, but now because we don't mint using the PSM currently here, agree it should just revert in that instance so the user won't be short changed